src/providers/mod.rs

```diff
@@ -0,0 +1,112 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::{
+  boxed::Box,
+  collections::HashMap,
+  error::Error,
+  fmt::{self, Display, Formatter},
+};
+
+use self::{mock::MockProvider as Mock, ses::SesProvider as Ses};
+use settings::Settings;
+
+mod mock;
+mod ses;
+
+trait Provider
+{
+  fn send(
+    &self,
+    to: String,
+    cc: Vec<String>,
+    subject: String,
+    body_text: String,
+    body_html: Option<String>,
+  ) -> Result<String, ProviderError>;
+}
```

Comment:
Obviously the idea behind this trait is to localise all provider-specific stuff inside the appropriate module and just expose a generic interface to the rest of the code.